CromIAM/src/main/scala/cromiam/sam/SamClient.scala

```diff
@@ -136,15 +136,17 @@ class SamClient(scheme: String,
     val createCollection = registerCreation(user, collection, cromIamRequest)
 
     createCollection flatMap {
-      case r if r.status == StatusCodes.Conflict => requestAuth(CollectionAuthorizationRequest(user, collection, "add"), cromIamRequest)
       case r if r.status == StatusCodes.NoContent => Monad[FailureResponseOrT].unit
       case r => FailureResponseOrT[IO, HttpResponse, Unit](IO.raiseError(SamRegisterCollectionException(r.status)))
+    } recoverWith {
+      case r if r.status == StatusCodes.Conflict => requestAuth(CollectionAuthorizationRequest(user, collection, "add"), cromIamRequest)
```

Comment:
Is this creating an infinite retry loop? Do we want a break-clause in here?

Reply:
Unless I've made a cut-paste errors, I believe these are still two different calls, where the latter never calls the former, thus no looping:

- The outer call is still `requestSubmission`, aka CREATE
- This inner call is still `requestAuth`, aka UPDATE

https://github.com/broadinstitute/cromwell/blob/852a330a6a3ad3b06e839112b752bdbfd0efe2a8/CromIAM/src/main/scala/cromiam/sam/SamClient.scala#L125-L129